examples/tutorials/pipeline_end_to_end_tutorial.ipynb

```diff
@@ -0,0 +1,1472 @@
+    "    - [Modeling](#Modeling)\n",
+    "    - [Pipeline Building](#Pipeline-Building)\n",
+    "    - [Running Pipelines Locally](#Running-Pipelines-Locally)\n",
+    "- [Recap](#Recap)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e6dbaf9",
+   "metadata": {},
+   "source": [
+    "## Intro"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "627bce95",
+   "metadata": {},
+   "source": [
+    "### Scenario\n",
+    "\n",
+    "As notebooks become more mature, they may get used like pipelines. For instance, notebooks that do the following things:\n",
+    "\n",
+    "* Process latest data and update dashboards/graphs/charts/reports\n",
+    "* Preprocess new data and dump it to the filesystem for downstream model development\n",
+    "* Create updated versions of models using latest data\n",
+    "* Run inference on new data using standard model\n",
+    "\n",
+    "may be expected to be re-executed as a pipeline on a regular or ad-hoc basis.\n",
+    "\n",
+    "However, you may not have the proper engineering support to set up these pipelines for production.\n",
+    "\n",
+    "### What might happen next?\n",
+    "\n",
+    "If this is the case, any of the following may happen:\n",
+    "\n",
+    "* You spend a lot of time copying and pasting and cleaning your code to make your pipeline work in orchestration systems or job schedulers (cron, Apache Airflow, Prefect, etc.).\n",
+    "* It takes forever to make your pipeline work and you end up running your pipelines manually whenever you need to (or get asked to).\n",
+    "* Your runtime environment is changed due to other experiments you are running and your pipeline is no longer working.\n",
+    "* You make some changes to your pipeline and now you find out you need to restart the above processes again.\n",
+    "\n",
+    "As more notebooks and pipelines need to be maintained, data scientists will need to spend more time on the operation side to make sure every notebook is running at the right time successfully. \n",
+    "This operational burden will consume a lot of time from the data scientist, which decreases productivity and innovation.\n",
+    "\n",
+    "### How can LineaPy help here?\n",
+    "\n",
+    "LineaPy can help to set up and maintain pipelines with minimal effort.\n",
+    "\n",
+    "```\n",
+    "import lineapy\n",
+    "\n",
+    "........................\n",
+    ".\n",
+    ". your original notebook\n",
+    ". \n",
+    "........................\n",
+    "\n",
+    "lineapy.save(object, 'artifact name')\n",
+    "lineapy.to_pipeline(['artifact name', 'other artifact name', ... ], framework, pipeline_name, output_dir)\n",
+    "```\n",
+    " \n",
+    "With just these three lines of code, LineaPy enables data scientists to produce runnable pipelines.\n",
+    "For some orchestration systems, like Apache Airflow, LineaPy is even able to upload the runnable pipeline without any manual interaction."
+   ]
```

Comment:
Add a line that leads up to a demo, something like "The rest of this demo notebook gives an example of how this can be done"